fractal-puppy/Engine
# Bug/Moving a weapon inside inventory unequips weapon fix

**Before:** If you had equipped a weapon and dragged another one from an inventory slot to another the weapon equipped got unequipped.
![ezgif-4-c35b2ec02d65](https://user-images.githubusercontent.com/17391875/65592297-99c9cc80-df8e-11e9-80d2-dada93a9da77.gif)


**Now:** It doesn't get unequipped.

To fix this bug important changes have been made to InventoryScript. Now pointers are used instead of copies of the Item object, so now instead of comparing by name, it is compared by the address in memory. Comparing by name could lead to many errors and also it is faster.

**To test:**
1. Open ItemsTest scene.
2. Hit play and pick up both weapons found on the floor.
3. Equip one weapon, then drag and drop the other weapon to any inventory slot.
4. Check that the player maintains the equipped weapon in hand and stats haven't changed.
